drake/solvers/trajectoryOptimization/PsmTraj.m

```diff
@@ -0,0 +1,445 @@
+classdef PsmTraj < NonlinearProgram
+  % Pseudospectral method
```

Comment:
Why `PsmTraj` is not a subclass of `DirectTrajectoryOptimization`? It fits the `DirectTrajectoryOptimization` model, and shares a lot of the functions. Also consider to rename it to `PseudoSpectralMethodTrajOpt`. `PSM` is not a common abbreviation.